### ISSUE TITLE
workload/tpcc: duplicate FK detection during initialization broken

### ISSUE THREAD

**Issue by nvanbenschoten** [cockroachdb/cockroach#37590]
The following TPCC roachtests have been failing for the past 3 days:
- #37250
- #37488
- #37519
- #36697
- #37163
- #37308
- #37549

The tests are now failing with the following error while loading their fixtures:
```
Error: restoring fixture: PostLoad hook: pq: column "ol_d_id" cannot be used by multiple foreign key constraints
```

This is almost certainly related to #37433, which was merged 4 days ago and touched some of the related logic in `create_table.go`. That change seems to be interacting poorly with duplicate foreign key detection here:
https://github.com/cockroachdb/cockroach/blob/58c567a325056033b326cb9c4ed9ba490e8956da/pkg/workload/tpcc/tpcc.go#L277-L280

The detection in `workload/tpcc` is regrettably using string matching instead of checking the pg error code. We should change it to search for `CodeInvalidForeignKeyError` instead.

@lucy-zhang The fix for this in `tpcc.go` is very straightforward and I'm happy to make that change, but before doing so, I want to make sure this isn't detecting an actual issue or unexpected change in behavior. A quick look over [3c49a5f](https://github.com/cockroachdb/cockroach/commit/3c49a5f22be27dd0770b9b1918b82a9e0b13d195#diff-10ff214b79444a5cefaec9397b2f0a28L579) doesn't show any deliberate reason why the error string returned here would have changed. Was this change in behavior expected?

NOTE: Most of these issues had previous failures, so once this problem is fixed, we should go back and minimize the comments about this (those in the past 3 days) instead of closing the issues entirely.

**Comment by lucy-zhang** [comment]
This is happening because #37433 (somewhat unintentionally) fixed a bug in foreign key validation. The bug was that we would previously allow things like
```
CREATE TABLE t (a INT, b INT, c INT, PRIMARY KEY (a, b), INDEX (c, b))
ALTER TABLE t ADD FOREIGN KEY (a, b) REFERENCES ...
ALTER TABLE t ADD FOREIGN KEY (c, b) REFERENCES ...
```
where one FK is on the primary key and the other is on some other index. We weren't including the PK when validating uniqueness of columns in FK constraints across a table, so `b` gets used twice.

The column that's being used twice here is `ol_d_id` (in the code path where we load the old incorrect tpcc fixtures with the wrong index). Prior to #37433, we would create the second FK that uses that column, which could have led to undefined behavior, since we assume elsewhere that each column has at most one FK constraint on it.

It's not clear to me how to fix these tests. My understanding is that the tpcc roachtests rely on the old incorrect test fixtures with both FK constraints on `ol_d_id`, but now it's forbidden to create the second FK constraint at all. So we'd either have to remove this FK from the tests, or re-generate the fixtures. @nvanbenschoten thoughts?

**Comment by lucy-zhang** [comment]
Also, to clarify, we weren't actually hitting the hardcoded `columns cannot be used by multiple foreign key constraints` error before. I'm not sure if we ever were, or what the expected behavior actually was.

**Comment by nvanbenschoten** [comment]
Got it, thanks for looking into this @lucy-zhang. As we discussed offline, `ol_d_id` should never have been allowed as the source of two different foreign keys, which would have avoided [this complication](https://github.com/cockroachdb/cockroach/commit/18d0b4284cca5b26ffbfc237f158a03df16a51ad). Now that we're correctly detecting the issue, we're running into problems with the invalid fixture.

So it sounds like the solution is to remove compatibility with fixtures that contain this incorrect index and foreign key. Conveniently, I already needed to regenerate fixtures for #36854, so I think it's time to just go ahead and do that. Doing so and merging #36854 will resolve this issue.
